lib/error-handler.js

```diff
@@ -40,19 +48,20 @@ export default function errorHandler (errorResponse) {
       payloadData: config.data
     }
   }
-
   if (isObject(data)) {
     if ('requestId' in data) {
       errorData.requestId = data.requestId
     }
-    if ('sys' in data && 'type' in data.sys) {
+    if ('message' in data) {
       errorData.message = data.message
-      if (data.details) {
-        errorData.details = data.details
-      }
     }
-    if ('sys' in data && 'id' in data.sys) {
-      errorName = data.sys.id
+    if ('details' in data) {
+      errorData.details = data.details
+    }
+    if ('sys' in data) {
+      if ('id' in data.sys) {
+        errorName = data.sys.id
```

Comment:
We could, but thats breaking the old behaviour:

`error.name = errorName && errorName !== 'Unknown' ? errorName : `${response.status} ${response.statusText}`